velox/vector/arrow/Bridge.cpp

```diff
@@ -1114,7 +1115,13 @@ VectorPtr importFromArrowImpl(
 
   if (arrowSchema.dictionary) {
     return createDictionaryVector(
-        pool, type, nulls, arrowSchema, arrowArray, isViewer, wrapInBufferView);
+        pool,
+        INTEGER(),
```

Comment:
We need to do some conversion if `type` is not `INTEGER`, cannot silently replace it.

Reply:
Does does this function even takes type as a parameter if it only accepts INTEGER?

Reply:
It can be expanded to support different widths of indices from arrow

Reply:
To make sure I am on same page: 
1. In arrow dictionaries can have different index types apart from integers ?
2. Since velox only supports integer indexes I will also make a change to createDictionary where we do the type conversion from whatever type arrow has to integer.

Reply:
This helps : https://arrow.apache.org/docs/format/CDataInterface.html#dictionary-encoded-arrays .

Reply:
FYI this diff doesnt add support for non integer index's - that should be in another diff.